src/runtime-rs/crates/hypervisor/src/qemu/inner.rs

```diff
@@ -270,6 +276,24 @@ impl QemuInner {
     }
 }
 
+async fn log_qemu_stderr(stderr: ChildStderr) -> Result<()> {
+    info!(sl!(), "starting reading qemu stderr");
+
+    let stderr_reader = BufReader::new(stderr);
+    let mut stderr_lines = stderr_reader.lines();
+
+    while let Some(buffer) = stderr_lines
+        .next_line()
+        .await
+        .context("next_line() failed on qemu stderr")?
+    {
+        info!(sl!(), "qemu stderr: {:?}", buffer);
```

Comment:
The equivalent code for `virtiofsd` also filters out lines that would only contain _whitespaces_. I'm not sure how this could happen but I guess it wouldn't harm to do the same here.

Reply:
I've never seen qemu log an empty line (although I haven't seen virtiofsd to do that either :-)).  I also believe that logging empty lines would better be fixed in the program that does that. ;-)

Reply:
> I've never seen qemu log an empty line (although I haven't seen virtiofsd to do that either :-)). I also believe that logging empty lines would better be fixed in the program that does that. ;-)

As any other command, QEMU is allowed to format the error message it prints on stderr as it wants.
These error messages are meant for a human user and can legitimately contain multiple lines and empty lines.

The code in this PR is collecting the QEMU stderr on a per-line basis and turns each line into a full shim log entry, i.e. a given QEMU error can span over multiple shim logs. We cannot do much about that but we could maybe avoid generating a log entry with no information in it... No big deal. We can still revisit this later, based on experience.